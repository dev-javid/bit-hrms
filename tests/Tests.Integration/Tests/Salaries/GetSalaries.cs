using Xunit.Abstractions;

namespace Tests.Integration.Tests.Salaries
{
    public class GetSalaries(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/salaries";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data_For_Supplied_Month_If_Month_Supplied()
        {
            // Arrange
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Salaries/GetSalaries.sql");

            // Act
            using HttpResponseMessage response = await Client.GetAsync($"{Route}?Month=6");

            // Assert
            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result.Ignore("items.[*].salaryId"));
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Date_For_Current_Month_If_Month_Not_Supplied()
        {
            // Arrange
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/Salaries/GetSalaries.sql");

            // Act
            using HttpResponseMessage response = await Client.GetAsync(Route);

            // Assert
            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result.Ignore("items.[*].salaryId", "items.[*].disbursedOn", "items.[*].month", "items.[*].year"));
        }
    }
}
