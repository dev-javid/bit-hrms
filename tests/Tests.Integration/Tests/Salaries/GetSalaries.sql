INSERT INTO salaries (company_id, employee_id, month, year, amount, created_by, created_on) 
VALUES 
(999, 1, EXTRACT(MONTH FROM CURRENT_DATE), EXTRACT(YEAR FROM CURRENT_DATE),  decode('r9a0jnes21bjwubV9WWhyD471a+GfkfaSxfWvm/EipI=', 'base64'), 1, NOW()),
(999, 2, EXTRACT(MONTH FROM CURRENT_DATE), EXTRACT(YEAR FROM CURRENT_DATE),  decode('NCreznXjp8Mj+MhlL7E1VJCTGtJ6yHRnlMW3SslYM74=', 'base64'), 1, NOW()),
(999, 1, 6,                                2024,                             decode('OANFUK+Ot3vm9O7Aav81agrBBURnBQovD5N29lTHRWs=', 'base64'), 1, '2024-01-01 00:00:00+05:30'),
(999, 2, 6,                                2024,                             decode('l8MD4C6izZebR2+62C191mDNZbTeknGpQNfchAnu1Po=', 'base64'), 1, '2024-01-01 00:00:00+05:30');

