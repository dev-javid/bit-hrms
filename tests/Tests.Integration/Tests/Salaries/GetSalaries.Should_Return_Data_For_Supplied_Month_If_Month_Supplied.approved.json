{
    "page": 1,
    "limit": 10,
    "total": 2,
    "items": [
        {
            "salaryId": "__IGNORED_VALUE__",
            "employeeId": 1,
            "compensation": 0,
            "amount": 6000,
            "month": 6,
            "year": 2024,
            "disbursedOn": "2023-12-31",
            "deductions": []
        },
        {
            "salaryId": "__IGNORED_VALUE__",
            "employeeId": 2,
            "compensation": 0,
            "amount": 10000,
            "month": 6,
            "year": 2024,
            "disbursedOn": "2023-12-31",
            "deductions": []
        }
    ]
}